sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -572,6 +572,10 @@ class Dataset[T] private[sql](
     val parsedDelay =
       Option(CalendarInterval.fromString("interval " + delayThreshold))
         .getOrElse(throw new AnalysisException(s"Unable to parse time delay '$delayThreshold'"))
+    // Threshold specified in months/years is non-deterministic
```

Comment:
what does "safe" mean in this context?  users must not rely on watermarks for correctness as they can be arbitrarily delayed based on batch boundaries.  I think this error actually confuses the point as its is enforcing precision when this API cannot provide that.